pkg/testsuite/mock/wallet_transactions.go

```diff
@@ -301,6 +380,169 @@ func (w *Wallet) TransitionImplicitAccountToAccountOutput(transactionName string
+	}
+	serialNumber := inputAccount.FoundryCounter + 1
+
+	totalIn := inputState.BaseTokenAmount()
+	baseTokenEach := totalIn / iotago.BaseToken(nNativeTokens+1)
+	remainder := totalIn - baseTokenEach*iotago.BaseToken(nNativeTokens+1)
+
+	tokenScheme := &iotago.SimpleTokenScheme{
+		MintedTokens:  big.NewInt(int64(nNativeTokens)),
+		MeltedTokens:  big.NewInt(0),
+		MaximumSupply: big.NewInt(1000),
+	}
+	nativeTokenFeature := &iotago.NativeTokenFeature{
+		ID:     lo.PanicOnErr(iotago.FoundryIDFromAddressAndSerialNumberAndTokenScheme(accountAddr, serialNumber, iotago.TokenSchemeSimple)),
+		Amount: big.NewInt(1),
+	}
+
+	outputStates = append(outputStates,
+		builder.NewFoundryOutputBuilder(accountAddr, baseTokenEach+remainder, serialNumber, tokenScheme).
+			MustBuild(),
+	)
+
+	outputStates = append(outputStates,
+		builder.NewAccountOutputBuilderFromPrevious(inputAccount).
+			FoundriesToGenerate(1).
+			MustBuild(),
+	)
+
+	for _, index := range addressIndexes {
+		outputStates = append(outputStates, &iotago.BasicOutput{
+			Amount: baseTokenEach,
+			Mana:   0,
+			UnlockConditions: iotago.BasicOutputUnlockConditions{
+				&iotago.AddressUnlockCondition{Address: w.Address(index)},
+			},
+			Features: iotago.BasicOutputFeatures{
+				nativeTokenFeature,
+			},
+		})
+	}
+
+	accountOutput, isAccount := outputStates[1].(*iotago.AccountOutput)
+	if !isAccount {
+		panic("output with alias accountName is not *iotago.AccountOutput")
+	}
+
+	signedTransaction := w.createSignedTransactionWithOptions(
+		transactionName,
+		[]uint32{0},
+		WithInputs(utxoledger.Outputs{inputState, inputAccountState}),
+		WithOutputs(outputStates),
+		WithBlockIssuanceCreditInput(&iotago.BlockIssuanceCreditInput{
+			AccountID: accountOutput.AccountID,
+		}),
+		WithCommitmentInput(&iotago.CommitmentInput{
+			CommitmentID: w.Node.Protocol.Engines.Main.Get().SyncManager.LatestCommitment().Commitment().MustID(),
+		}),
+	)
+
+	return signedTransaction
+}
+
+func (w *Wallet) TransitionFoundry(transactionName string, inputName string, accountName string) *iotago.SignedTransaction {
+	input := w.Output(inputName)
+	inputFoundry, isFoundry := input.Output().(*iotago.FoundryOutput)
+	if !isFoundry {
+		panic(fmt.Sprintf("output with alias %s is not *iotago.FoundryOutput", inputName))
+	}
+	inputAccount := w.AccountOutput(accountName)
+	nativeTokenAmount := inputFoundry.FeatureSet().NativeToken().Amount
+	previousTokenScheme, isSimple := inputFoundry.TokenScheme.(*iotago.SimpleTokenScheme)
+	if !isSimple {
+		panic("only simple token schemes supported")
+	}
+	tokenScheme := &iotago.SimpleTokenScheme{
+		MaximumSupply: previousTokenScheme.MaximumSupply,
+		MeltedTokens:  previousTokenScheme.MeltedTokens,
+		MintedTokens:  previousTokenScheme.MintedTokens.Add(previousTokenScheme.MintedTokens, big.NewInt(1)),
+	}
+
+	outputFoundry := builder.NewFoundryOutputBuilderFromPrevious(inputFoundry).
```

Comment:
We are actually already using the nativeTokenAmount from an existing foundry output and increasing it by one. But what we should check here is that the max supply is not already minted as this would cause the tx to be invalid.